auslib/test/blobs/test_systemaddons.py

```diff
@@ -222,3 +222,26 @@ def testDoesNotContainForbiddenDomain(self):
 """)
         self.assertFalse(blob.containsForbiddenDomain('gg',
                                                       self.whitelistedDomains))
+
+    def testIncorrectAddonLayout(self):
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+    "name": "fake",
+    "schema_version": 1000,
+    "hashFunction": "SHA512",
+    "addons": {
+        "c": {
+            "version": "1",
+            "platforms": {
+                "p": {
+                    "filesize": 2,
+                    "hashValue": "3",
+                    "fileUrl": "http://a.com/blah"
+                }
+            }
+        }
+    }
+}
+""")
+        blob.validate('gg', self.whitelistedDomains)
```

Comment:
It looks like the blob is invalid because "uninstall" is not provided at the top level, which is specified as required at https://github.com/tyagi-iiitv/balrog/blob/1936206e36d970889871581141cf4f13e965bdd1/auslib/blobs/schemas/systemaddons.yml#L10.